metagraph/core/plugin.py

```diff
@@ -283,9 +308,17 @@ def compute_concrete_properties(cls, obj, props: List[str]) -> Dict[str, Any]:
 
         The properties are cached to speed up future calls for the same properties.
         """
+        assert cls.is_typeclass_of(
+            obj
+        ), f"Cannot compute concrete properties of {obj} using {cls}"
+
         if len(props) == 0:
             return {}
 
+        # Be forgiving if a single string is passed in
```

Comment:
Changing the type hint from `props: Set[str]` to `props: Union[str, Set[str]]` might be a good idea here as well.